File: lib/greenwich/angle.rb
module Greenwich
  #===========================================================================
  # Module for angles
  #===========================================================================
  module Angle
    module_function

    #-------------------------------------------------------------------------
    # Normalize angle into the range 0 <= a < 2pi.
    #
    # @param:  angle  (Before normalized)
    # @return: angle  (Normalized angle)
    #-------------------------------------------------------------------------
    def norm_angle(angle)
      while angle <   0; angle += PI2; end
      while angle > PI2; angle -= PI2; end
      return angle
    rescue => e
      raise
    end
  end
end

File: lib/angle.py
"""
Module for angles
"""
from lib import const as cst


def norm_angle(angle):
    """ Normalize angle into the range 0 <= a < 2pi.

    :param  float angle: Before normalized
    :return float angle: Normalized angle
    """
    try:
        while angle < 0:
            angle += cst.PI2
        while angle > cst.PI2:
            angle -= cst.PI2
        return angle
    except Exception as e:
        raise

